### PR TITLE
feat(appo11y): add services operations for per-endpoint RED breakdown

### PR DESCRIPTION
**Pull request by cedricziel**
## Summary

- Adds \`gcx appo11y services operations <name>\` — pairs with \`services get\` to drill into a single service: one row per \`span_name\` with rate, error rate / percent, avg latency, p50/p95/p99, and a **time-share %** column (share of the service's total wall-clock time each operation consumes).
- Rows sorted by time-share desc, so low-rate-high-latency operations correctly outrank high-rate-fast ones for "where the time is spent" analysis. Concretely: 1 req/s × 2s outranks 100 req/s × 5ms in this view, even though the latter is 100× higher rate.
- \`--limit 15\` default (with the same truncation-hint pattern as \`services list\`), \`--limit 0\` for unlimited. The first stack I tested has services emitting 9,771 distinct \`span_name\` values — the default keeps output to a screenful.

## Example output

```
OPERATION                                       RATE         ERROR %  P95       TIME %
GET                                             4.084 req/s  0.00%    192.86ms  69.82%
POST                                            0.110 req/s  0.00%    6.962s    8.40%
POST /api/checkout                              0.022 req/s  0.00%    712.50ms  4.58%
POST /api/cart                                  0.088 req/s  0.00%    235.00ms  4.24%
GET /api/cart                                   0.082 req/s  0.00%    232.32ms  3.71%
GET /                                           0.083 req/s  0.00%    197.50ms  2.05%
...
```

\`-o wide\` adds P50, P99, and absolute ERRORS.

## What's shared with \`services get\`

- Bare-name resolution via \`target_info\` (ambiguity errors out the same way).
- Metrics-mode auto-detect + \`--metrics-mode {auto,v3,tempo,otel}\` override.
- \`--since\` (PromQL duration), \`--kind\`, \`--namespace\`, \`--datasource\`, \`-o\`.
- \`fail.NewCommandUsageError\` wrapping for option validation.
- Exit-code-1 on no-telemetry, snapshot still emitted so agents see the shape.

## What's new

- Extended the per-mode \`metricNames\` table with \`latencyCount\` and \`latencySum\` (needed for avg latency, which the time-share metric requires).
- Four per-operation PromQL builders: \`buildOperationsRateQuery\`, \`buildOperationsErrorRateQuery\`, \`buildOperationsLatencyQuantileQuery\`, \`buildOperationsAvgLatencyQuery\` — all \`by (span_name)\`.
- \`mergeOperations\` pivots six per-operation maps into one row per \`span_name\` and computes time-share in a single pass.
- New table codec with two column tiers (default / wide), backed by \`style.NewTable\` so it shares the same Neon-Dark styling as \`services list\`.

## Test plan

- [x] Unit tests cover all four query builders across all three metric families, the per-\`span_name\` extractor, \`mergeOperations\` (including time-share ranking, no-avg-latency edge case, zero-traffic edge case, error-percent + has-errors inference), opts validation, and codec rendering (default columns, wide columns, empty path, wrong-type rejection).
- [x] \`mise run lint\` clean.
- [x] \`GCX_AGENT_MODE=false go test ./...\` passes.
- [x] Reference docs regenerated.
- [x] Live verification against a Grafana Cloud stack:
  - default table for a high-traffic service returns sensible time-share ranking (high-rate \`GET\` dominates, but slow \`POST /api/checkout\` correctly appears above faster routes with similar rate).
  - \`-o wide\` adds the extra columns cleanly.
  - JSON shape: \`{service, window, metrics_mode, span_kinds, items[]}\`.
  - Ambiguous bare name errors out (\`flagd\` exists in two namespaces on the test stack).
  - \`--metrics-mode tempo\` override returns rate from the legacy series but \`has_avg=false\` / \`has_latency_*=false\` (same predicted mismatch as \`services get\`).
  - Nonexistent service exits 1.
  - \`--limit 0\` returns the full 9,771 ops for the high-cardinality frontend.

## Follow-ups

- \`services map <name>\` — service-graph slice (callers + callees), still on the original plan.
- The list papercuts Igor flagged (\`--namespace\` filter and "non-empty env" filter on \`services list\`).

### CODE REVIEW

File: docs/reference/cli/gcx_appo11y_services_operations.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+Break down a service's RED into per-operation rows (span_name × rate/errors/latency).
+
+### Synopsis
+
+List the per-operation RED breakdown for one service.
+
+The argument is either the bare service name or the canonical
+"<namespace>/<name>" form; bare names are auto-resolved against
+target_info the same way "gcx appo11y services get" does it.
```
Comment (dafydd-t) [inline docs/reference/cli/gcx_appo11y_services_operations.md]:
```suggestion
target_info the same way as "gcx appo11y services get".
```

File: docs/reference/cli/gcx_appo11y_services_operations.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+Rows are sorted by time-share desc — the share of the service's total
+wall-clock time each operation consumes, computed as
+(avg_latency × rate) / sum(all). This surfaces operations that dominate
+latency regardless of whether they're high-rate-fast or low-rate-slow.
+
+The span-metric family is auto-detected by default; see
+"gcx appo11y services get --help" for the --metrics-mode reference.
```
Comment (dafydd-t) [inline docs/reference/cli/gcx_appo11y_services_operations.md]: this doesnt make a whole lot of sense to me - am i just missing context? I don't see any reference to `span-metric family` in a table output

File: docs/reference/cli/gcx_appo11y_services_operations.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+```
+gcx appo11y services operations <service> [--namespace ns] [flags]
+```
+
+### Examples
+
+```
+
+  # Top operations for a service in the default 5m window
```
Comment (dafydd-t) [inline docs/reference/cli/gcx_appo11y_services_operations.md]:
```suggestion
  # Top operations for a service called "checkout" in the default 5m window
```

Comment (dafydd-t) [inline docs/reference/cli/gcx_appo11y_services_operations.md]:
```suggestion
  # Top operations for a service called "checkout" in the default 5m window
```

File: docs/reference/cli/gcx_appo11y_services_operations.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+## gcx appo11y services operations
```
Comment (dafydd-t) [inline docs/reference/cli/gcx_appo11y_services_operations.md]: I would prefer a verb for the name, rather than `operations` here. even `get-operations` would read more clearly for me. Do you have any suggestions?

File: docs/reference/cli/gcx_appo11y_services_operations.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  gcx appo11y services operations payments/checkoutservice --since 1h --limit 0 -o json
+```
+
+### Options
+
+```
+  -d, --datasource string     Prometheus datasource UID (defaults to datasources.prometheus in config or auto-discovery)
+  -h, --help                  help for operations
+      --json string           Comma-separated list of fields to include in JSON output, or 'list' (or '?') to discover available fields
+      --kind string           Span kinds to include: inbound (server+consumer), server, consumer, all, or a comma list of SPAN_KIND_* literals (default "inbound")
```
Comment (dafydd-t) [inline docs/reference/cli/gcx_appo11y_services_operations.md]:
```suggestion
      --kind string           Span kinds to include. Choose either: inbound (server+consumer), server, consumer, all, or a comma separated list of SPAN_KIND_* literals (default "inbound")
```

File: docs/reference/cli/gcx_appo11y_services_operations.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      --limit int             Limit the number of operations returned (0 = unlimited; applied after sorting by time-share desc) (default 15)
+      --metrics-mode string   Span-metrics family: auto (probes the stack), v3 (traces_span_metrics_*), tempo (traces_spanmetrics_*), or otel (bare calls_total + duration_seconds_bucket) (default "auto")
```
Comment (dafydd-t) [inline docs/reference/cli/gcx_appo11y_services_operations.md]:
```suggestion
      --metrics-mode string   Span-metrics family. Choose either: auto (probes the stack), v3 (traces_span_metrics_*), tempo (traces_spanmetrics_*), or otel (bare calls_total + duration_seconds_bucket) (default "auto")
```

Comment (dafydd-t) [reply docs/reference/cli/gcx_appo11y_services_operations.md]: Or `One of` or something, like you have below

**Review by dafydd-t — COMMENTED** [review]
nice! It works as you describe, i just have a comment about the command name, and some nits about the help docs which you can decide whether to include or not

**Comment by wardbekker** [comment]
## Shepherd Status

Rebased this branch onto latest `main` and resolved the conflicts that made it unmergeable (it collided with the recently-merged `services map` / `services get` work):

- **`query.go`** — additive merge: kept both the service-graph map functions and the new operations functions (git had textually interleaved two distinct functions that share a loop skeleton).
- **`commands.go`** — registered both `map` and `operations` subcommands.
- **`formatPercentMaybe`** — this helper was defined in both `map.go` (main) and `operations.go` (this PR). Consolidated it into `get.go` alongside the sibling formatters and removed both copies.

Verified locally: build ✓, `golangci-lint` ✓ (0 issues), full `-race` test suite ✓, generated docs regenerate with zero drift. Force-pushed (`--force-with-lease`); a fresh CI run is in progress.

### Open items left for the author (@cedricziel)
- **Command name** — @dafydd-t suggested a verb-based name (e.g. `get-operations`) over `operations`. Left as-is for you to decide.
- **Help-text nits** — @dafydd-t's optional wording suggestions on the `--kind` / `--metrics-mode` descriptions and the example comment are not applied (they're tied to the naming choice).

---
*Updated by shepherd-pr*

**Comment by cedricziel** [comment]
Thanks for the review! Addressed everything and rebased on `main` (resolved conflicts with the now-merged `services map` work). Force-pushed.

**Command name** — renamed `operations` → `list-operations`. You were right that it should be a verb; per CONSTITUTION § CLI Grammar (`$AREA $NOUN $VERB`) `operations` was a bare noun. I went with `list-operations` over `get-operations` because it returns a list (one row per `span_name`), so it parallels the sibling `list` rather than the single-resource `get`.

**Help/doc nits** — all applied:
- `target_info the same way as "gcx appo11y services get".`
- Reworded the confusing "span-metric family" sentence to name the actual series it auto-detects (`traces_spanmetrics_*` / `traces_span_metrics_*` / bare `calls_total`) and how to pin it.
- Example comment now names the service.
- `--kind` / `--metrics-mode`: rephrased to `One of: …` (consistent with the `--output` help below) and `comma-separated`. Applied the same wording to `services get` too so the two stay aligned.

Reference docs regenerated, `mise run all` green.

**Review by dafydd-t — APPROVED** [review]
LGTM! thanks a lot @cedricziel !
